app-creator/client/src/widgets/app-root.ts

```diff
@@ -93,19 +93,59 @@ export class AppRoot extends LitElement {
       font-weight: 500;
       color: var(--accent-color);
     }
+
+    paper-progress {
+      width: 100%;
+      height: 20px;
+      z-index: 10;
+      position: absolute;
+      top: 0;
+      left: 0;
+      margin: 0px;
+    }
   `;
 
   /**
    * Sets template dialog search query.
    */
   @property({ type: String }) query = '';
 
+  /**
+   * Array of templates.
+   */
+  @property({ type: Array }) templates: TemplateItem[] = [];
+
+  /**
+   * Loading state.
+   */
+  @property({ type: Boolean }) loading = false;
+
   /**
    * Reference to dialog element.
    */
   @query('paper-dialog') dialog!: PaperDialogElement;
 
+  async fetchTemplates() {
+    try {
+      this.loading = true;
+
+      const response = await fetch('/templates');
+      const templates = await response.json();
+
+      templatesManager.setTemplates(templates);
+      this.templates = templatesManager.getTemplates();
```

Comment:
We are storing the templates in a TemplatesManager instance so we don't have to fetch the templates again when rendering the templates-tab. We can just get the already fetched templates. One missing addition is limiting the number of fetched templates. We would need to fetch ~5 templates at first, and then add a `View more templates` button for fetching more templates.